geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync, unlinkSync } from 'fs';
+import { Extract } from 'unzipper';
+const mapshaper = require('mapshaper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          Extract({
+            path: `${fileInfo.destination}/${fileInfo.filename.replace(
+              '.zip',
+              '',
+            )}`,
```

Comment:
to be platform agnostic you may use `import * as path from 'path'` and `path.join(fileInfo.destination, path.basename(fileInfo.filename))`